hw4/01-routing.js

```diff
@@ -12,38 +12,58 @@ const port = process.env.PORT || 5000;
 
 // For other routes, such as http://localhost:5000/other, this exercise should return a status code 404 with '404 - page not found' in html format
 
-const routes = [
-  'welcome',
-  'redirect',
-  'redirected',
-  'cache',
-  'cookie',
-  'other',
-];
+const routes = ['welcome', 'redirect', 'redirected', 'cache', 'cookie', 'other'];
 
 let getRoutes = () => {
-  let result = '';
+    let result = '';
+    routes.forEach((elem) => (result += `<li><a href="/${elem}">${elem}</a></li>`));
+    return result;
+};
 
-  routes.forEach(
-    (elem) => (result += `<li><a href="/${elem}">${elem}</a></li>`)
-  );
+app.get('/', function (req, res) {
+    let routeResults = getRoutes();
+    res.writeHead(200, { 'Content-Type': 'text/html' });
+    res.write(`<h1>Exercise 04</h1>`);
+    res.write(`<ul> ${routeResults} </ul>`);
+    res.end();
+});
 
-  return result;
-};
+app.get('/welcome', function (req, res) {
+    res.writeHead(200, { 'Content-Type': 'text/html' });
+    res.write(`<h1>Welcome Here!</h1>`);
+    res.end();
+});
 
-app.get('/', (req, res) => {
-  let routeResults = getRoutes();
+app.get('/redirect', function (req, res) {
+    res.writeHead(302, { Location: '/redirected' });
+    res.end();
+});
 
-  res.writeHead(200, { 'Content-Type': 'text/html' });
-  res.write(`<h1>Exercise 04</h1>`);
-  res.write(`<ul> ${routeResults} </ul>`);
-  res.end();
+app.get('/redirected', function (req, res) {
+    res.writeHead(200, { 'Content-Type': 'text/html' });
+    res.write(`<h1>You are in the redirected page!</h1>`);
+    res.end();
 });
 
-app.get('/welcome', (req, res) => {});
+app.get('/cache', function (req, res) {
+    res.writeHead(200, { 'Content-Type': 'text/plain', 'Cache-Control': 'max-age=86400' });
+    res.write('This resource was cached!');
+    res.end();
+});
+
+app.get('/cookie', function (req, res) {
```

Comment:
I ran your code and tried to go to `localhost.com:5001/cookie` but got this error: `Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client`